src/github/projectKit.ts

```diff
@@ -109,39 +164,56 @@ export class ProjectKit implements IProjectApi {
     await this.gql(UPDATE_PROJECT_V2_ITEM_FIELD_VALUE_QUERY, { project, item, targetField, targetFieldValue });
   }
 
-  async assignIssueToProject(issue: Issue, projectId: string): Promise<boolean> {
+  async assignIssueToProject(issue: Issue, projectId: string): Promise<string> {
     try {
       const migration = await this.gql<{ addProjectV2ItemById: { item: { id: string } } }>(
         ADD_PROJECT_V2_ITEM_BY_ID_QUERY,
         { project: projectId, issue: issue.node_id },
       );
 
-      return !!migration.addProjectV2ItemById.item.id;
+      return migration.addProjectV2ItemById.item.id;
     } catch (e) {
       this.logger.error("Failed while executing 'ADD_PROJECT_V2_ITEM_BY_ID_QUERY' query");
       throw e;
     }
   }
 
-  async addIssueToProject(issue: Issue, project: NodeData): Promise<boolean> {
-    this.logger.info(`Syncing issue #${issue.number} for ${project.title}`);
+  async fetchProjectFieldNodeValues(project: NodeData, projectFields?: FieldValues): Promise<FieldValues> {
+    if (!projectFields) {
+      throw new Error("'projectsFields' is null!");
+    }
 
-    return await this.assignIssueToProject(issue, project.id);
-  }
+    const projectFieldData = await this.getProjectFields(project.id);
 
-  async assignIssue(issue: Issue): Promise<boolean> {
-    const project = await this.fetchProjectData();
+    const { field, value } = projectFields;
 
-    return await this.addIssueToProject(issue, project);
+    // ? Should we use .localeCompare here?
+    const customField = projectFieldData.find(({ name }) => name.toUpperCase() === field.toUpperCase());
 
-    // TODO: Assign targetField
-  }
+    // check that this custom field exists and it has available options to set up
+    if (!customField) {
+      throw new Error(`Field ${field} does not exist!`);
+    } else if (!customField.options) {
+      throw new Error(`Field ${field} does not have any available options!.` + "Please add options to set values");
```

Comment:
What if this custom field is of text type?
It will not have any options, right?
Or do we support only fields with options for now?

<img width="537" alt="Screenshot 2023-01-23 at 13 01 17" src="https://user-images.githubusercontent.com/12039224/214034892-c95ec8ad-8591-46f0-a3b4-09cbf7df5960.png">

Reply:
Another reason for which I wouldn't trouble myself too much on this is because GitHub [has a working implementation](https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/using-the-built-in-automations) similar to this one.

They should provide better support for those particular cases.

GitHub Issue Sync is used more to move an issue state when new labels are added, which is the feature they do not provide.

Reply:
Sounds good. Consider having something like:

```
if (!customField) {
  throw new Error(xxx);
} else if (customField.type !== "Single select") {
  throw new Error(`Field ${field} is of type ${customField.type}. Only Single select is supported.`)
} else if (!customField.options) {
  throw new Error(xxx)
```

IMO it would give more precise info for the user.

Reply:
We do not handle types in the implementation I'm afraid.

I call the [change field](https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/using-the-api-to-manage-projects#updating-a-single-select-field) from GraphQL which handles that change, but I don't have the type (I haven't figure out how to get it yet).

The array that I can work with is: 
```json
[
  {
    "id": "PVTF_lADOBHQkoc4AKgznzgGi63Y",
    "name": "Title"
  },
  {
    "id": "PVTF_lADOBHQkoc4AKgznzgGi63c",
    "name": "Assignees"
  },
  {
    "id": "PVTSSF_lADOBHQkoc4AKgznzgGi63g",
    "name": "Status",
    "options": [
      {
        "id": "f75ad846",
        "name": "Todo"
      },
      {
        "id": "47fc9ee4",
        "name": "In Progress"
      },
      {
        "id": "98236657",
        "name": "Done"
      }
    ]
  },
  {
    "id": "PVTF_lADOBHQkoc4AKgznzgGi63k",
    "name": "Labels"
  },
  {
    "id": "PVTF_lADOBHQkoc4AKgznzgGi63o",
    "name": "Linked pull requests"
  },
  {
    "id": "PVTF_lADOBHQkoc4AKgznzgGi63w",
    "name": "Reviewers"
  },
  {
    "id": "PVTF_lADOBHQkoc4AKgznzgGi630",
    "name": "Repository"
  },
  {
    "id": "PVTF_lADOBHQkoc4AKgznzgGi634",
    "name": "Milestone"
  }
]
```

And I get that array from [here](https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/using-the-api-to-manage-projects#finding-the-node-id-of-a-field)

Reply:
I see. Ok, all makes sense now.